### PR TITLE
Codechef COINS code in Python with Readme (Issue #4682)

### PR DESCRIPTION
**Pull request by ka1shi**
**Fixes issue:**
#4682 

**Changes:**
Added python solution for Codechef Bytelandian gold coins (COINS) Problem
Added Readme file for that problem

### CODE REVIEW

File: code/online_challenges/src/codechef/COINS/COINS.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+a = {};
+
```
Comment (arnavb) [inline code/online_challenges/src/codechef/COINS/COINS.py]: Reformat this file using black.

File: code/online_challenges/src/codechef/COINS/COINS.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    if n in a:
+        return a[n];
+        
+    a[n] = max(n, dollar(int(n/2))+dollar(int(n/3))+dollar(int(n/4)));
+    return a[n];
+
+while(True):
+    try:
+        n = int(input());
+    except EOFError:
```
Comment (arnavb) [inline code/online_challenges/src/codechef/COINS/COINS.py]: No need for the try catch.

File: code/online_challenges/src/codechef/COINS/COINS.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        a[n]=n;
+        return a[n];
+        
+    if n in a:
+        return a[n];
+        
+    a[n] = max(n, dollar(int(n/2))+dollar(int(n/3))+dollar(int(n/4)));
+    return a[n];
+
+while(True):
```
Comment (arnavb) [inline code/online_challenges/src/codechef/COINS/COINS.py]: No need for parentheses.

File: code/online_challenges/src/codechef/COINS/COINS.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    try:
+        n = int(input());
+    except EOFError:
+        break;
+        
+    if(n<=11):
+        print(n);
+    else:
+        print(dollar(n));
+
```
Comment (arnavb) [inline code/online_challenges/src/codechef/COINS/COINS.py]: Only one trailing newline at the end of the file.

File: code/online_challenges/src/codechef/COINS/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# Problem Link:
+[COINS] (https://www.codechef.com/problems/COINS)
```
Comment (arnavb) [inline code/online_challenges/src/codechef/COINS/README.md]:
```suggestion
[COINS](https://www.codechef.com/problems/COINS)
```

File: code/online_challenges/src/codechef/COINS/COINS.py (hunk 1 above)
Comment (ka1shi) [reply code/online_challenges/src/codechef/COINS/COINS.py]: Done

File: code/online_challenges/src/codechef/COINS/COINS.py (hunk 2 above)
Comment (ka1shi) [reply code/online_challenges/src/codechef/COINS/COINS.py]: @arnavb, the input contains test cases that will continue till End of File, there is no predefined number of inputs. So to catch that error, we need to use try and catch. Please suggest me if anything else can be done to improve. Thank You!

File: code/online_challenges/src/codechef/COINS/COINS.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+a = {}
+
+def dollar(n):
+
```
Comment (arnavb) [inline code/online_challenges/src/codechef/COINS/COINS.py]:
```suggestion
```

**Pull request merged**